module/module.py

```diff
@@ -1044,7 +1053,56 @@ def get_ui_external_links(self):
         return lst
 
     def get_search_string(self, default=""):
-        return self.request.query.get('search', default)
+        """Manage the query parameter `search`
+
+        The business impact logic behind this:
+        - the UI defined BI (problems_business_impact) is considered first
+        - the user minimum BI is then considered
+
+        Thus, the maximum value of both BIs is added as a filter in the search query if no
+        BI is specified in the search query. If more than one bi: exist in the search query
+        the last one is the only one that is kept in the query parameter.
+        """
+        if not self.request.route.config.get('search_engine', None):
+            # Requested URL do not include the search engine
+            logger.debug("No search engine on this page!")
+            return ''
+
+        search_string = self.request.query.get('search', default)
+
+        # Force include the UI BI in the search query if it is not yet present
+        bi = self.problems_business_impact
+        if "bi:" not in search_string:
+            # Include BI as the first search criterion
+            return ("bi:>=%d " % bi) + search_string
+
+        # Search patterns like: isnot:0 isnot:ack isnot:"downtime fred" name "vm fred"
+        regex = re.compile(SEARCH_QUERY_PATTERNS, re.VERBOSE)
+        logger.debug("Search string: %s", search_string)
+        new_ss = ''
+        for match in regex.finditer(search_string):
+            if not match.group('key'):
+                continue
+            if match.group('key') not in ['bp', 'bi']:
+                logger.debug("- match: %s / %s", match.group('key'), match.group('value'))
+                new_ss += match.group('key')
+                new_ss += match.group('value')
+                continue
+            logger.debug("Found BI filter: %s", match.group('value'))
+            s = match.group('value')
+            bi = self.problems_business_impact
+            try:
+                if s.startswith('>=') or s.startswith('<='):
+                    bi = int(s[2:])
+                elif s.startswith('>') or s.startswith('<') or s.startswith('='):
+                    bi = int(s[1:]) + 1
+            except ValueError:
+                pass
+            logger.debug("- search BI is >=%d", bi)
+        new_ss = ("bi:>=%d " % bi) + new_ss
+        logger.debug("- updated search string: %s", new_ss)
+
+        return new_ss
```

Comment:
This is not the right place to do this. « get_search_string » should only return the search string. You need to add all this where the search string is generated. It is too late to compute anything here, the "search_string" variable is just use as a placeholder.

Reply:
I do not agree with this. There is not only one place where the search string is built ... it may come from the form, but it can also be a direct query parameter. This functino is called when and where the search string query parameter is interesting or the applicatino. So it is, IMO, the right place to do this stuff.

Reply:
I keep my position, for the same reasons stated above. I will not accept this patch.